cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/JsonMenu/utilities.ts

```diff
@@ -58,10 +63,87 @@ function getJSONConfig(widgetJSONData) {
   return config;
 }
 
+function parsePluginJSON(filename, pluginJSON) {
+  // Parse filename in order to set pluginName and pluginType
+  // Currently the filename is designed to be <pluginName>-<pluginType>.json
+  const [pluginName, pluginType] = filename.split('-');
+
+  // Parse file data in order to populate the rest of properties
+  const basicPluginInfo = {
+    // If the string fields are undefined, set them to empty string
+    displayName: pluginJSON['display-name'] ? pluginJSON['display-name'] : '',
+    pluginName: pluginName ? pluginName : '',
+    pluginType: pluginType ? pluginType : '',
+    emitAlerts: pluginJSON['emit-alerts'],
+    emitErrors: pluginJSON['emit-errors'],
+  } as IBasicPluginInfo;
+
+  const newConfigurationGroups = [];
+  const newGroupToInfo = {};
+  const newGroupToWidgets = {};
+  const newWidgetToInfo = {};
+  const newWidgetToAttributes = {};
+
+  pluginJSON['configuration-groups'].forEach((groupObj) => {
```

Comment:
I believe we already have this logic right? https://github.com/cdapio/cdap/blob/develop/cdap-ui/app/cdap/components/ConfigurationGroup/utilities/index.ts#L93 Can we extract this and make it generic and re-use this?

Reply:
As per discussion yesterday, I'll be extracting this in the https://github.com/cdapio/cdap/pull/12244.